frontend/src/board/pgn/boardTools/underboard/settings/GameSettings.tsx

```diff
@@ -162,20 +233,31 @@ const SaveGameButton = ({
         request.reset();
     };
 
-    const onSave = (headers?: GameHeader) => {
+    const onSave = (newHeaders?: GameHeader) => {
         request.onStart();
 
+        if (!newHeaders && headersChanged) {
+            newHeaders = {
+                white: headers.White || '?',
+                black: headers.Black || '??',
+                result: headers.Result,
+                date: headers.Date,
+            };
+        }
+
         const update: UpdateGameRequest = {
             orientation,
-            unlisted:
-                (visibility === 'unlisted') === game.unlisted
-                    ? undefined
-                    : isFreeTier || visibility === 'unlisted',
+            unlisted: unlisted,
+            publish: isPublishing,
             timelineId: game.timelineId,
         };
 
-        if (headers) {
-            update.headers = headers;
+        if (newHeaders) {
+            for (const [name, value] of Object.entries(newHeaders)) {
+                chess?.setHeader(name.charAt(0).toUpperCase() + name.slice(1), value);
```

Comment:
I'd prefer to just change the field names in `GameHeader` to `White`, `Black`, etc instead of doing this string manipulation

Reply:
:+1: I agree what I have here is a bit excessive. Thank you.